pkg/kubelet/types/pod_update.go

```diff
@@ -104,9 +104,14 @@ func GetPodSource(pod *api.Pod) (string, error) {
 type SyncPodType int
 
 const (
+	// SyncPodSync is when the pod is synced to ensure desired state
 	SyncPodSync SyncPodType = iota
+	// SyncPodUpdate is when the pod is updated from source
 	SyncPodUpdate
+	// SyncPodCreate is when the pod is created from source
 	SyncPodCreate
+	// SyncPodKill is when the pod is killed based on a trigger internal to the kubelet.
+	SyncPodKill
```

Comment:
caveat: pod workers don't rely on the SyncPodType to do any real work in syncPod at all, except for the new `SyncPodKill`. Might warrant a comment here.